File: src/main/java/LDS/Person/service/WikiNewService.java
package LDS.Person.service;

import LDS.Person.dto.request.WikiNewCreateRequest;
import LDS.Person.dto.request.WikiNewPageQueryRequest;
import LDS.Person.dto.request.WikiNewReviewRequest;
import LDS.Person.dto.response.WikiNewResponse;
import LDS.Person.dto.response.WikiNewListResponse;
import LDS.Person.dto.response.WikiNewReviewResponse;
import LDS.Person.dto.response.PageResponse;

/**
 * Wiki 新增业务逻辑接口
 */
public interface WikiNewService {

  /**
   * 新增 Wiki 记录
   * 
   * @param request 创建请求
   * @return Wiki 响应
   */
  WikiNewResponse createWikiNew(WikiNewCreateRequest request);

  /**
   * 分页查询 Wiki 新增列表
   * 支持按 wiki_states 和 wikinew_id 过滤
   * 
   * @param request 分页查询请求
   * @return 分页响应
   */
  PageResponse<WikiNewListResponse> pageQuery(WikiNewPageQueryRequest request);

  /**
   * 根据 Wiki ID 查询完整内容
   * 
   * @param wikinewId Wiki ID
   * @return Wiki 响应，如果不存在则返回 null
   */
  WikiNewResponse getWikiNewById(Long wikinewId);

  /**
   * 检查 Wiki 键名是否已存在
   * 
   * @param keyName Wiki 键名
   * @return true 表示键名已存在，false 表示不存在
   */
  boolean isKeyNameExists(String keyName);

  /**
   * 审核 Wiki 新增申请
   * 批准（wikiStates=1）时将内容复制到 wiki 表
   * 驳回（wikiStates=2）时仅更新状态
   * 
   * @param request 审核请求
   * @return 审核响应
   */
  WikiNewReviewResponse reviewWikiNew(WikiNewReviewRequest request);
}

File: src/Services/WikiNewService.cs
using SidusX.DTOs.Request;
using SidusX.DTOs.Response;
using SidusX.Models;
using SidusX.Repositories;

namespace SidusX.Services;

public class WikiNewService
{
    private readonly WikiNewRepository _repo;
    private readonly WikiRepository _wikiRepo;
    public WikiNewService(WikiNewRepository repo, WikiRepository wikiRepo) { _repo = repo; _wikiRepo = wikiRepo; }

    public async Task<WikiNewResponse> CreateAsync(WikiNewCreateRequest req)
    {
        if (string.IsNullOrWhiteSpace(req.KeyName)) throw new ArgumentException("key_name 不能为空");
        if (string.IsNullOrWhiteSpace(req.Texts)) throw new ArgumentException("texts 不能为空");
        if (await _wikiRepo.IsKeyNameExistsAsync(req.KeyName)) throw new ArgumentException($"wiki中已有此key_name: {req.KeyName}");
        if (await _repo.IsKeyNameExistsAsync(req.KeyName)) throw new ArgumentException($"wiki_new中已有此key_name: {req.KeyName}");
        var now = DateTime.Now;
        var n = new WikiNew { KeyName = req.KeyName, Texts = req.Texts, Tags = req.Tags, Version = 1.0, CreateTime = now, CreateUser = req.CreateUser, UpdateTime = now, UpdateUser = req.CreateUser, WikiStates = 0 };
        n.WikinewId = await _repo.InsertAsync(n);
        return ToResponse(n);
    }

    public async Task<WikiNewResponse?> GetByIdAsync(long id)
    {
        var n = await _repo.GetByIdAsync(id);
        return n == null ? null : ToResponse(n);
    }

    public async Task<PageResponse<WikiNewListResponse>> PageQueryAsync(WikiNewPageQueryRequest req)
    {
        var (items, total) = await _repo.PageQueryAsync(req);
        return new PageResponse<WikiNewListResponse>
        {
            Total = total, Page = req.Page, PageSize = req.PageSize,
            Data = items.Select(n => new WikiNewListResponse { WikinewId = n.WikinewId, KeyName = n.KeyName, WikiStates = n.WikiStates, CreateTime = n.CreateTime, CreateUser = n.CreateUser }).ToList()
        };
    }

    public async Task<WikiNewReviewResponse?> ReviewAsync(WikiNewReviewRequest req)
    {
        var n = await _repo.GetByIdAsync(req.WikinewId);
        if (n == null) return null;
        await _repo.UpdateStateAsync(req.WikinewId, req.WikiStates);
        // If approved, insert into wiki
        if (req.WikiStates == 1)
        {
            var now = DateTime.Now;
            var wiki = new Wiki { KeyName = n.KeyName, Texts = n.Texts, Tags = n.Tags, Version = 1.0, CreateTime = now, CreateUser = n.CreateUser, UpdateTime = now, UpdateUser = n.CreateUser };
            await _wikiRepo.InsertAsync(wiki);
        }
        return new WikiNewReviewResponse { WikinewId = req.WikinewId, WikiStates = req.WikiStates, Message = req.WikiStates == 1 ? "审核通过，已创建Wiki" : "已拒绝" };
    }

    private static WikiNewResponse ToResponse(WikiNew n) => new()
    {
        WikinewId = n.WikinewId, KeyName = n.KeyName, Texts = n.Texts, Tags = n.Tags,
        Version = n.Version, CreateTime = n.CreateTime, CreateUser = n.CreateUser,
        UpdateTime = n.UpdateTime, UpdateUser = n.UpdateUser, WikiStates = n.WikiStates
    };
}
